databases/week3/dbweek3.sql

```diff
@@ -0,0 +1,261 @@
+DEFAULT CHARACTER SET = 'utf8mb4';
+
+USE  mealsharingdata;
+
+CREATE TABLE `meal` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `title` varchar(255) NOT NULL,
+  `description` text NULL DEFAULT NULL, 
+  `location` varchar(255) NOT NULL ,
+  `when` varchar(255) NULL DEFAULT NULL,
+  `max_reservations`  int(10) NULL DEFAULT NULL,
+  `price` decimal(5,2) NOT NULL,
+  `created_date` DATE NOT NULL
+);
+
+CREATE TABLE `reservation` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `number_of_guests` int(32) NOT NULL,
+  `meal_id` int(10) unsigned NOT NULL,
+  `created_date` DATE NOT NULL, 
+  `contact_phonenumber` varchar(255) NOT NULL,
+  `contact_name` varchar(255) NOT NULL,
+  `contact_email` varchar(255) NULL DEFAULT NULL,
+  CONSTRAINT `fk_meal` FOREIGN KEY (`meal_id`) REFERENCES `meal` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+);
+
+CREATE TABLE `review` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `title` varchar(255) NOT NULL,
+  `description` text(10) NULL DEFAULT NULL,
+  `review_meal_id` int(10) unsigned NOT NULL, 
+  `stars` int(10) unsigned NOT NULL,
+  `created_date` DATE NOT NULL,
+  CONSTRAINT `fk_review_meal_id` FOREIGN KEY (`review_meal_id`) REFERENCES `meal` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+);
+
+-- Inserts to meal table
+INSERT INTO 
+  meal(title, description, location, price, created_date)
+VALUES
+  ('Carbonara', 'Designating pasta that is mixed, just after boiling, with a sauce of sautéed bacon, 
+  Parmesan cheese, pepper, and beaten eggs, formed as the eggs are cooked slightly by the hot pasta.', ' Holsteinsgade 1, 8300 Odder',
+   144.50, '2021-10-10');
+
+INSERT INTO 
+  meal(title, description, location, price, created_date)
+VALUES
+  ('Avocado toast', 'Avocado toast can be a nutritious breakfast, as avocados are a good source of healthy fats and very filling.', ' Holsteinsgade 1, 8660 Skanderborg',
+   70.00, '2021-10-13');
+
+INSERT INTO 
+  meal(title, description, location, price, created_date)
+VALUES
+  ('Sticky and Spicy Baked Cauliflower', 'Battering and oven-baking large cauliflower florets gives them a slightly crackly, just-rich-enough exterior that’s ideal for coating in a sticky-sweet gochujang glaze.', ' Holsteinsgade 1, 8660 Skanderborg',
+   170.40, '2021-10-14');
+
+INSERT INTO 
+  meal(title, description, location, price, created_date, max_reservations)
+VALUES
+  ('Greek salat', 'Popular salad generally made with pieces of tomatoes, cucumbers, onion, feta cheese, and olives and dressed with salt, pepper, Greek oregano, and olive oil.', ' M. P. Bruuns Gade 25, 8000 Aarhus',
+   100.10, '2021-10-11', 3);
+
+INSERT INTO 
+  meal(title, description, location, price, created_date, max_reservations)
+VALUES
+  ('Vanilla Cake', 'A classic butter cake but with Japanese techniques applied for the most plush, soft and moist yellow cake like you’ve never had before.', ' M. P. Bruuns Gade 25, 8000 Aarhus',
+   60.50, '2021-09-11', 5);
```

Comment:
Could have used 1 INSERT INTO statement will many values () ,
For the records here you don't have a value for "max_reservations" for example, you could pass Null.

Reply:
@mateusms01  Hi. Thank you for your review. What do you mean about  "1 INSERT INTO" ?
I mentioned  NULL by default for "max_reservations"; so that's why I do not have a value in INSERT. Is is right?

Reply:
What I meant is that you don't need to write one insert statement for each record you want to insert. 
Its more direct and clear. For example:

INSERT INTO meal(title, description, location, price, created_date, max_reservations)
VALUES 
('Carbonara', 'Designating pasta that is mixed, just after boiling, with a sauce of sautéed bacon, 
  Parmesan cheese, pepper, and beaten eggs, formed as the eggs are cooked slightly by the hot pasta.', ' Holsteinsgade 1, 8300 Odder',  144.50, '2021-10-10', Null),
  ('Avocado toast', 'Avocado toast can be a nutritious breakfast, as avocados are a good source of healthy fats and very filling.', ' Holsteinsgade 1, 8660 Skanderborg', 70.00, '2021-10-13', Null), 
  ('Sticky and Spicy Baked Cauliflower', 'Battering and oven-baking large cauliflower florets gives them a slightly crackly, just-rich-enough exterior that’s ideal for coating in a sticky-sweet gochujang glaze.', ' Holsteinsgade 1, 8660 Skanderborg', 170.40, '2021-10-14', Null),
  ('Greek salat', 'Popular salad generally made with pieces of tomatoes, cucumbers, onion, feta cheese, and olives and dressed with salt, pepper, Greek oregano, and olive oil.', ' M. P. Bruuns Gade 25, 8000 Aarhus', 100.10, '2021-10-11', 3);